src/main/java/fi/okm/jod/yksilo/service/profiili/YksiloService.java

```diff
@@ -21,7 +29,43 @@
 public class YksiloService {
   private final YksiloRepository yksilot;
 
-  public void deleteYksilo(JodUser user) {
+  public Yksilo get(JodUser user) {
+    return yksilot
+        .findById(user.getId())
+        .orElseThrow(() -> new NotFoundException("Profiili does not exist"));
+  }
+
+  public void update(
+      JodUser user, YksiloDto dto, Authentication authentication, HttpServletRequest request) {
+    var yksilo =
+        yksilot
+            .findById(user.getId())
+            .orElseThrow(() -> new NotFoundException("Profiili does not exist"));
+    var tervetuloapolku = dto.tervetuloapolku() != null ? dto.tervetuloapolku() : false;
+    yksilo.setTervetuloapolku(tervetuloapolku);
+    yksilot.save(yksilo);
+
+    // Update the authentication object in the session
+    var oldAuthentication = (Saml2Authentication) authentication;
+    var principal = (JodSaml2Principal) oldAuthentication.getPrincipal();
+    SecurityContextHolder.getContext()
+        .setAuthentication(
+            new Saml2Authentication(
+                new JodSaml2Principal(
+                    principal.getName(),
+                    principal.getAttributes(),
+                    principal.getSessionIndexes(),
+                    principal.getRelyingPartyRegistrationId(),
+                    principal.getId(),
+                    tervetuloapolku),
+                oldAuthentication.getSaml2Response(),
+                oldAuthentication.getAuthorities()));
+    request
+        .getSession()
+        .setAttribute("SPRING_SECURITY_CONTEXT", SecurityContextHolder.getContext());
```

Comment:
I think that we should not re-create the authentication object here. This also breaks under the mock authentication since it assumes Saml authentication.  We could change the spring.session.redis.save-mode to on_get_attribute to force it to save the new value, but an authentication object should be immutable, and that would also cause unnecessary writes into the cache (the attribute is read on every request).

I'd change the implementation so that we just use YksiloDto and (optionally) cache it into the session as a separate attribute. That also makes it possible to extend the parameters without touching the authentication implementation.